proto/wippersnapper/i2c/v1/i2c.proto

```diff
@@ -205,3 +207,122 @@ message I2CDeviceEvent {
+  }
+}
+
+///*** I2C Output Device Messages (from i2c_output.proto in api v2) ***///
+
+/**
+* LedBackpackBlinkRate represents supported, OPTIONAL,
+* blink rates for LED backpack displays
+*/
+enum LedBackpackBlinkRate {
+  LED_BACKPACK_BLINK_RATE_UNSPECIFIED = 0; /** No blinking. **/
+  LED_BACKPACK_BLINK_RATE_OFF         = 1; /** No blinking. **/
+  LED_BACKPACK_BLINK_RATE_2HZ         = 2; /** 2 Hz blink rate. **/
+  LED_BACKPACK_BLINK_RATE_1HZ         = 3; /** 1 Hz blink rate. **/
+  LED_BACKPACK_BLINK_RATE_HALFHZ      = 4; /** 0.5 Hz blink rate. **/
+}
+
+/**
+* LedBackpackAlignment represents all text alignment
+* options for LED backpack displays
+*/
+enum LedBackpackAlignment {
+  LED_BACKPACK_ALIGNMENT_UNSPECIFIED = 0; /** Unspecified alignment option. **/
+  LED_BACKPACK_ALIGNMENT_LEFT        = 1; /** (Default) Left-aligned. **/
+  LED_BACKPACK_ALIGNMENT_RIGHT       = 2; /** Right-aligned. **/
+}
+
+/**
+* Desired SSD1306 text 'magnification' size.
+*/
+enum Ssd1306TextSize {
+  SSD1306_TEXT_SIZE_UNSPECIFIED = 0; /** Unspecified text size. **/
+  SSD1306_TEXT_SIZE_1           = 1; /** Default text size. 6x8. **/
+  SSD1306_TEXT_SIZE_2           = 2; /** Text size 2. 12x16. **/
+  SSD1306_TEXT_SIZE_3           = 3; /** Text size 3. 18x24. **/
+}
+
+/**
+* LedBackpackConfig represents the configuration for a LED backpack display.
+*/
+message LedBackpackConfig {
+  int32 brightness                = 1; /** Desired brightness of the LED backpack, from 0 (off) to 15 (full brightness). **/
+  LedBackpackAlignment alignment  = 2; /** Desired text alignment for the LED backpack. **/
+}
+
+/**
+* CharLCDConfig represents the configuration for a character LCD display.
+*/
+message CharLCDConfig {
+  uint32 rows            = 1; /** Number of rows for the character LCD. **/
+  uint32 columns         = 2; /** Number of columns for the character LCD. **/
+  string backlight_color = 3 [(nanopb).max_size = 15]; /** Optional Backlight color for the character LCD, in Hex. **/
+}
+
+/**
+* Ssd1306Config represents the configuration for a SSD1306 OLED display.
+*/
+message Ssd1306Config {
+  uint32 width              = 1; /** Width of the display. **/
+  uint32 height             = 2; /** Height of the display. **/
+  Ssd1306TextSize text_size = 3; /** Desired text 'magnification' size. **/
+}
+
+/**
+* I2cOutputAdd represents a request from the broker to add an I2C output device to a device.
+*/
+message I2cOutputAdd {
+  oneof config {
+    LedBackpackConfig led_backpack_config = 1; /** Configuration for LED backpack. **/
+    CharLCDConfig char_lcd_config         = 2; /** Configuration for character LCD. **/
+    Ssd1306Config ssd1306_config          = 3; /** Configuration for SSD1306 OLED display. **/
+  }
+}
+
+/**
+* LedBackpackWrite represents a request from the broker to write a message to a LED backpack.
+*/
+message LedBackpackWrite {
```

Comment:
Inconsistent naming schemes SSSD/LED
Remove all the per-device Write messages
Switch that to the write command, feed it a string
Remove anything unused, for now

Reply:
I don't want to remove the per-device write messages in favor of a generalized `string message` in case we'll eventually support i2c output devices that don't only print strings